RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -0,0 +1,1030 @@
+  const unsigned int nv = y.getSize();
+  for (unsigned int k = 0; k < nv; k++)
+     if ( edm::isNotFinite(y.pk_[k]) || edm::isNotFinite(y.z_[k]) ) { y.pk_[k]=0; y.z_[k]=0;}
+
+  for (unsigned int i = 0; i < nt; i++) // initialize
+    tks.Z_sum_[i] = rho0 * local_exp(-beta * dzCutOff_ * dzCutOff_);
+
+  // improve vectorization (does not require reduction ....)
+  for (unsigned int k = 0; k < nv; k++) {
+     for (unsigned int i = 0; i < nt; i++)  
+       tks.Z_sum_[i] += y.pk_[k] * local_exp(-beta * Eik(tks.z_[i], y.z_[k],tks.dz2_[i], tks.t_[i], y.t_[k],tks.dt2_[i]));
+  }
+
+
+  for (unsigned int k = 0; k < nv; k++) {
+    GlobalPoint pos(0, 0, y.z_[k]);
+    
+    vector<reco::TransientTrack> vertexTracks;
+    for (unsigned int i = 0; i < nt; i++) {
+      if (tks.Z_sum_[i] > 1e-100) {
+	
+	double p = y.pk_[k] * local_exp(-beta * Eik(tks.z_[i], y.z_[k], tks.dz2_[i],
+                                                    tks.t_[i], y.t_[k], tks.dt2_[i] )) / tks.Z_sum_[i];
+	if ((tks.pi_[i] > 0) && (p > mintrkweight_)) {
+	  vertexTracks.push_back(*(tks.tt[i]));
+	  tks.Z_sum_[i] = 0; // setting Z=0 excludes double assignment
+	}
+      }
+    }
+    TransientVertex v(pos, dummyError, vertexTracks, 0);
+    clusters.push_back(v);
+  }
+
+  return clusters;
+
+}
+
+vector<vector<reco::TransientTrack> > DAClusterizerInZT_vect::clusterize(
+		const vector<reco::TransientTrack> & tracks) const {
+  
+#ifdef VI_DEBUG
+  if (verbose_) {
+    std::cout  << "###################################################" << endl;
+    std::cout  << "# vectorized DAClusterizerInZT_vect::clusterize   nt=" << tracks.size() << endl;
+    std::cout  << "###################################################" << endl;
+  }
+#endif
+  
+  vector<vector<reco::TransientTrack> > clusters;
+  vector<TransientVertex> && pv = vertices(tracks);
+  
+#ifdef VI_DEBUG
+  if (verbose_) {
+    std::cout  << "# DAClusterizerInZT::clusterize   pv.size=" << pv.size()
+					     << endl;
+  }
+#endif
+
+  if (pv.empty()) {
+    return clusters;
+  }
+  
+  // fill into clusters and merge
+  vector<reco::TransientTrack> aCluster = pv.begin()->originalTracks();
+  
+  for (auto k = pv.begin() + 1; k != pv.end(); k++) {
+    if ( std::abs(k->position().z() - (k - 1)->position().z()) > (2 * vertexSize_)) {
+      // close a cluster
+      if (aCluster.size()>1){
+	clusters.push_back(aCluster);
+      }else{
+#ifdef VI_DEBUG
+	if(verbose_){
+	  std::cout << " one track cluster at " << k->position().z() << "  suppressed" << std::endl;
+	}
+#endif
+      }
+      aCluster.clear();
+    }
+    for (unsigned int i = 0; i < k->originalTracks().size(); i++) {
+      aCluster.push_back(k->originalTracks()[i]);
+    }
+    
+  }
+  clusters.emplace_back(std::move(aCluster));
+  
+  return clusters;
+
+}
+
+
+
+void DAClusterizerInZT_vect::dump(const double beta, const vertex_t & y,
+				  const track_t & tks, int verbosity) const {
+#ifdef VI_DEBUG
```

Comment:
is there a good reason to fold ::dump with ```#ifdef VI_DEBUG ``` ?

Reply:
I see that it was triggered by https://github.com/cms-sw/cmssw/pull/19935#issuecomment-326655958
Uhm, OK, but not very practical.

Reply:
otherwise there are a bunch of couts compiled in the regular release, which we don't want